src/Core/Framework/Test/Api/Controller/HealthCheckControllerTest.php

```diff
@@ -0,0 +1,50 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\Test\Api\Controller;
+
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Framework\Api\HealthCheck\Event\HealthCheckEvent;
+use Shopware\Core\Framework\Test\TestCaseBase\AdminFunctionalTestBehaviour;
+use Shopware\Core\Framework\Test\TestCaseHelper\CallableClass;
+
+/**
+ * @internal
+ */
+class HealthCheckControllerTest extends TestCase
+{
+    use AdminFunctionalTestBehaviour;
+
+    public function testHealthyInstance(): void
+    {
+        $this->getBrowser()->request('GET', '/api/_info/health-check');
+        $response = $this->getBrowser()->getResponse();
+
+        static::assertEquals(200, $response->getStatusCode());
+    }
+
+    public function testHealthCheckEventIsDispatched(): void
+    {
+        $listener = $this->getMockBuilder(CallableClass::class)->getMock();
```

Comment:
You could use `\Shopware\Core\Test\Stub\EventDispatcher\AssertingEventDispatcher` - see `\Shopware\Tests\Integration\Core\Content\Media\Infrastructure\Path\MediaLocationBuilderTest::testBuildMedia` for an example